networkx/generators/geometric.py

```diff
@@ -19,8 +19,8 @@
 ]
 
 
-@nx._dispatch(preserve_node_attrs={"G": {"pos": None}})
-def geometric_edges(G, radius, p=2):
+@nx._dispatch(node_attrs="position")
+def geometric_edges(G, radius, position="pos", p=2):
```

Comment:
I think all the other geometric functions defined in the module call `_geometric_edges` internally, so a similar keyword argument would have to be added to them as well. It gets tricky there though because most of them already have a kwarg called `pos`, which is supposed to be a dictionary of positions keyed by node.

Perhaps we should call this something like `position_attr` instead and add it to the rest of the geometric graph generation functions?